core/src/main/java/org/elasticsearch/search/suggest/term/TermSuggestionBuilder.java

```diff
@@ -117,33 +150,59 @@ public TermSuggestionBuilder sort(String sort) {
      * n-grams.
      * </ol>
      */
-    public TermSuggestionBuilder stringDistance(String stringDistance) {
+    public TermSuggestionBuilder stringDistance(@Nullable StringDistanceImpl stringDistance) {
         this.stringDistance = stringDistance;
         return this;
     }
 
     /**
+     * Get the string distance implementation setting.
+     */
+    @Nullable
+    public StringDistanceImpl stringDistance() {
+        return stringDistance;
+    }
+
+    /**
      * Sets the maximum edit distance candidate suggestions can have in order to
      * be considered as a suggestion. Can only be a value between 1 and 2. Any
      * other value result in an bad request error being thrown. Defaults to
      * <tt>2</tt>.
      */
-    public TermSuggestionBuilder maxEdits(Integer maxEdits) {
+    public TermSuggestionBuilder maxEdits(int maxEdits) {
+        checkArgument(maxEdits > 0 && maxEdits < 3, "maxEdits");
```

Comment:
I like the succinctness of this checks, wondering if java 8 offers anything to avoid having to introduce the utility class for this though.